streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -702,7 +702,6 @@ public synchronized Topology addStateStore(final StoreBuilder storeBuilder,
      * @return itself
      * @throws TopologyException if the processor of state is already registered
      */
-    @SuppressWarnings("unchecked")
```

Comment:
@mjsax Can we tighten the parameters to `StoreBuilder<KeyValueStore>`? This would push the compiler warning into the user's code without actually breaking any compatibility.

My only hesitation is that I suspect that we did it this way because our store builders don't always return the parameterized type.

Reply:
I think, it could actually break a built if we add the generic. Thus, we can only do this, if it is considers a bug-fix. The original KIP does not propose to add a generic type though. Thus, I am not sure. \cc @guozhangwang @dguy 

IMHO, it might make sense to relax the constraint and fix `InternatlTopoogyBuilder` -- can you check if this can be done?

Reply:
I see. In that case, I'd prefer to leave it as-is for now. We can relax or restrict the type in a future KIP. I don't want the scope to creep on this PR.

Reply:
@guozhangwang , thanks for creating https://issues.apache.org/jira/browse/KAFKA-7075